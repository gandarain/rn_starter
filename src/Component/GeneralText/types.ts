export type Props = {
  title: string;
  size: number;
  color: string;
  variant: string;
};
