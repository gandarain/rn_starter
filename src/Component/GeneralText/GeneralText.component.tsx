import React from 'react';
import { Text } from 'react-native';

import config from './GeneralText.config';
import { Props } from './types';
import styles from './GeneralText.styles';

/**
 * GeneralText
 * @param {Props} props - props
 * @return {JSX.Element} - GeneralText Component
 * @constructor
 */
const GeneralText: React.FC<Props> = ({
  title,
  color,
  variant,
  size,
}: Props): JSX.Element => (
  <Text style={styles.title(color, variant, size)}>{title}</Text>
);

GeneralText.displayName = config.displayName;
GeneralText.defaultProps = config.defaultProps;

export default GeneralText;
