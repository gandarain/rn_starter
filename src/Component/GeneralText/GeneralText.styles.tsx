const styles = {
  title: (color: string, variant: string, size: number) => ({
    fontFamily: variant,
    fontSize: size,
    color,
  }),
};

export default styles;
