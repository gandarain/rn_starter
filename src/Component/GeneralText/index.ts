export { default } from './GeneralText.component';
