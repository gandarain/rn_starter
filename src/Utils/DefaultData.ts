/* eslint-disable require-jsdoc */
const emptyString = '';
const emptyObject = {};
const noop = () => {};

export { emptyObject, emptyString, noop };
