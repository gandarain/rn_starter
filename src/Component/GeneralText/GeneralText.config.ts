import Utils from '../../Utils';

const {
  DefaultData: { emptyString },
} = Utils;

const displayName = 'GeneralText';
const defaultProps = {
  title: emptyString,
  size: 0,
  color: emptyString,
  variant: emptyString,
};

export default { displayName, defaultProps };
