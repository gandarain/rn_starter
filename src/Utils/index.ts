// @flow
import * as DefaultData from './DefaultData';

export default {
  DefaultData,
};
