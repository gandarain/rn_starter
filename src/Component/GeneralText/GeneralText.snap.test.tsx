import * as React from 'react';
import renderer from 'react-test-renderer';

import Utils from '../../Utils';

const {
  DefaultData: { emptyString },
} = Utils;
import GeneralText from './GeneralText.component';

const defaultProps = {
  title: emptyString,
  size: 0,
  color: emptyString,
  variant: emptyString,
};

describe('<GeneralText />', () => {
  it('Should render GeneralText', () => {
    const tree = renderer.create(<GeneralText {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
