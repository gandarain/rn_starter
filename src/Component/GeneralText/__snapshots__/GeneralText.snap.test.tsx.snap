// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeneralText /> Should render GeneralText 1`] = `
<Text
  style={
    Object {
      "color": "",
      "fontFamily": "",
      "fontSize": 0,
    }
  }
>
  
</Text>
`;
